docs/source/testing_services.rst

```diff
@@ -0,0 +1,205 @@
+:mod:`ChildEmulator <octue.cloud.emulators.child.ChildEmulator>` constructor parameters. Here's an example JSON file
+with just the messages:
+
+.. code-block:: json
+
+    {
+        "messages": [
+            {
+                "type": "log_record",
+                "content": {"msg": "Starting analysis."}
+            },
+            {
+                "type": "log_record",
+                "content": {"msg": "Finished analysis."}
+            },
+            {
+                "type": "monitor_message",
+                "content": {"progress": "35%"}
+            },
+            {
+                "type": "result",
+                "content": {"output_values": [1, 2, 3, 4, 5], "output_manifest": null}
+            }
+        ]
+    }
+
+You can then instantiate a child emulator from this in python:
+
+.. code-block:: python
+
+    child_emulator = ChildEmulator.from_file("path/to/emulated_child.json")
+
+    def handle_monitor_message(message):
+        ...
+
+    result = child_emulator.ask(
+        input_values={"hello": "world"},
+        handle_monitor_message=handle_monitor_message,
+    )
+    >>> {"output_values": [1, 2, 3, 4, 5], "output_manifest": None}
+
+
+Using the child emulator
+------------------------
+To emulate your children in tests, patch the :mod:`Child <octue.resources.child.Child>` class with the
+:mod:`ChildEmulator <octue.cloud.emulators.child.ChildEmulator>` class.
+
+.. code-block:: python
+
+    from unittest.mock import patch
+
+    from octue import Runner
+    from octue.cloud.emulators import ChildEmulator
+
+
+    app_directory_path = "path/to/directory_containing_app"
+
+    # You can explicitly specify your children here as shown or
+    # read the same information in from your app configuration file.
+    children = [
+        {
+            "key": "my_child",
+            "id": "octue/my-child-service",
+            "backend": {
+                "name": "GCPPubSubBackend",
+                "project_name": "my-project"
+            }
+        },
+    ]
+
+    runner = Runner(
+        app_src=app_directory_path,
+        twine=os.path.join(app_directory_path, "twine.json"),
+        children=children,
+        service_id="you/your-service",
+    )
+
+    emulated_children = [
+        ChildEmulator(
+            id="octue/my-child-service",
+            internal_service_name="you/your-service",
+            messages=[
+                {
+                    "type": "result",
+                    "content": {
+                        "output_values": [300],
+                        "output_manifest": None,
+                    }
+                },
+            ]
+        )
+    ]
+
+    with patch("octue.runner.Child", side_effect=emulated_children):
+        analysis = runner.run(input_values={"some": "input"})
+
+
+Notes:
+- If your app uses more than one child, simply provide more child emulators in the ``emulated_children`` list in the
```

Comment:
I'm increasingly trying to remove words like "simply" and "just" in instructional contexts like this because they can be patronising (marketing content might be different, but that's a different context).